pandas/core/generic.py

```diff
@@ -5429,6 +5429,8 @@ def _reindex_with_indexers(
 
         if (copy or copy is None) and new_data is self._mgr:
             new_data = new_data.copy(deep=copy)
+        elif using_copy_on_write() and new_data is self._mgr:
+            new_data = new_data.copy(deep=copy)
```

Comment:
Could you also add an explicit test that covers this?

Reply:
I prefer my version, this is easier to read imo. The conditions are nicely aligned.

Reply:
The align copy false test covers this

Reply:
> Could you also add an explicit test that covers this?

Sorry, that wasn't clear ;) This change is in `_reindex_with_indexers`, which is used by align, but is also used through `reindex`? (I assumed that align would use it _through_ reindex, but that's actually not the case, it is using it directly) And so I was wondering if this change is also needed for correct behaviour of `reindex` (and so if we missed something there), and if so, it would be good to also have a test for `reindex` that covers this.

Reply:
Ah got you.
We can't get there through reindex, because we have early exits preventing us from reaching ``_reindex_with_indexers`` if both axes are equal to the object that is reindexed. 

I also looked into avoiding this function call when using align, but there is nothing to be gained there. It makes more sense covering this here.